storage/remote/client.go

```diff
@@ -74,13 +65,39 @@ func (c *Client) Store(samples model.Samples) error {
 		req.Timeseries = append(req.Timeseries, ts)
 	}
 
-	ctxt, cancel := context.WithTimeout(context.TODO(), c.timeout)
-	defer cancel()
+	data, err := proto.Marshal(req)
+	if err != nil {
+		return err
+	}
+
+	buf := bytes.Buffer{}
+	if _, err := snappy.NewWriter(&buf).Write(data); err != nil {
+		return err
+	}
 
-	_, err := c.client.Write(ctxt, req)
+	httpReq, err := http.NewRequest("POST", c.url, &buf)
 	if err != nil {
 		return err
 	}
+	httpReq.Header.Add("Content-Encoding", "snappy")
+	httpResp, err := c.client.Do(httpReq)
+	if err != nil {
+		return err
+	}
+	defer httpResp.Body.Close()
+	if httpResp.StatusCode/100 != 2 {
+		return fmt.Errorf("server returned HTTP status %s", httpResp.Status)
+	}
+
+	buf.Reset()
+	if _, err = buf.ReadFrom(snappy.NewReader(httpResp.Body)); err != nil {
```

Comment:
It seemed like the right thing to do?  This is what grpc was doing.

I guess its meaning less so we could get rid of it.  I don't think there are any implication for forwards compatibility.